src/Bicep.LangServer/Telemetry/BicepTelemetryEvent.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Bicep.LanguageServer.Telemetry
+{
+    public record BicepTelemetryEvent : TelemetryEventParams
+    {
+        public Dictionary<string, string> Properties { get; set; } = new Dictionary<string, string>();
+
+        public string? EventName { get; set; }
+
+        public static BicepTelemetryEvent Create(string eventName)
```

Comment:
I didn't explain myself well with the suggestion of implementing `Create` - what I was thinking was implementing a set of `Create` methods - e.g. something like:

```csharp
public record BicepTelemetryEvent : TelemetryEventParams
{
    public string? EventName { get; set; }

    public Dictionary<string, string>? Properties { get; set; }

    public static BicepTelemetryEvent CreateTopLevelDeclarationSnippetInsertion(string name)
        => new BicepTelemetryEvent {
            EventName = TelemetryConstants.EventNames.TopLevelDeclarationSnippetInsertion,
            Properties = new() {
                ["name"] = name,
            },
        };

    public static BicepTelemetryEvent CreateResourceBodySnippetInsertion(string name, string type)
        => new BicepTelemetryEvent {
            EventName = TelemetryConstants.EventNames.ResourceBodySnippetInsertion,
            Properties = new() {
                ["name"] = name,
                ["type"] = type,
            },
        };

    public static BicepTelemetryEvent CreateModuleBodySnippetInsertion(string name)
        => new BicepTelemetryEvent {
            EventName = TelemetryConstants.EventNames.ModuleBodySnippetInsertion,
            Properties = new() {
                ["name"] = name,
            },
        };
}
```

This has the benefit of grouping all the telemetry 'schemas' in one place, and simplifies the telemetry raising code.

Reply:
This is definitely cleaner. Thanks! Updated here:
https://github.com/Azure/bicep/commit/62d6763f06b3d53bb8189b517dab0567a37fd0bd8